api/interactives.go

```diff
@@ -193,19 +210,29 @@ func (api *API) UpdateInteractiveHandler(w http.ResponseWriter, req *http.Reques
 	}
 
 	WriteJSONBody(mergedMap, w, http.StatusOK)
-
 }
 
 func (api *API) ListInteractivesHandler(w http.ResponseWriter, req *http.Request, limit int, offset int) (interface{}, int, error) {
 	// fetches all/filtered visulatisations
 	ctx := req.Context()
-	datasets, totalCount, err := api.mongoDB.ListInteractives(ctx, offset, limit)
+	db, totalCount, err := api.mongoDB.ListInteractives(ctx, offset, limit)
 	if err != nil {
 		log.Error(ctx, "api endpoint getDatasets datastore.GetDatasets returned an error", err)
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return nil, 0, err
 	}
-	return datasets, totalCount, nil
+	response := make([]*interactives.Interactive, 0)
+	for _, interactive := range db {
+		i, err := models.ToRest(interactive)
+		if err != nil {
+			log.Error(ctx, "cannot map db to http response", err)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return nil, 0, err
+		}
+		response = append(response, i)
+	}
+
+	return response, totalCount, nil
```

Comment:
i think we need to look at the other responses too - they are not too restful .

Get - is just a map - should be the model.
Post - is just a string - should be model again imho.
 
But can sort later 😊

Reply:
With the metadata story, map will be a struct now :)